linode/manager
# feat: [M3-8936] - Add Date Presets Functionality to Date Picker component.

## Description 📝

Add Date Presets Functionality to Date Range Picker component.

## Changes  🔄

List any change(s) relevant to the reviewer.

- Add Date Presets to DateRangePicker component.
- Test coverage to DateRangePicker

## Target release date 🗓️
N/A

## Preview 📷


https://github.com/user-attachments/assets/c7a97e77-562c-497f-a174-dcf7d247d662



## How to test 🧪

### Verification steps

(How to verify changes)

- Checkout the branch and run locally 
- Import the DateRangePicker component in any CM page
- You can use below example usage
```bash
 <DateTimeRangePicker
        onChange={({ end, preset, start }) => {
        }}
        endDateProps={{ label: 'End Date and Time', value: null }}
        format="yyyy-MM-dd HH:mm"
        presetsProps={{ enablePresets: true }}
        startDateProps={{ label: 'Start Date and Time', value: null }}
      />
```

<details>
<summary> Author Checklists </summary>

## As an Author, to speed up the review process, I considered 🤔

👀 Doing a self review
❔ Our [contribution guidelines](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md)
🤏 Splitting feature into small PRs
➕ Adding a [changeset](https://github.com/linode/manager/blob/develop/docs/CONTRIBUTING.md#writing-a-changeset)
🧪 Providing/improving test coverage
 🔐 Removing all sensitive information from the code and PR description
🚩 Using a feature flag to protect the release
👣 Providing comprehensive reproduction steps
📑 Providing or updating our documentation
🕛 Scheduling a pair reviewing session
📱 Providing mobile support
♿  Providing accessibility support

<br/>


## As an Author, before moving this PR from Draft to Open, I confirmed ✅


</details>

---
